src/backend/controllers/mealsController.js

```diff
@@ -1,101 +1,175 @@
-    res.status(200).json(allMeals);
+
+    if ('maxPrice' in hasQuery) {
+      const maxPrice = hasQuery.maxPrice;
+      const cheapMeals = await knex
+        .select("*")
+        .from("meal")
+        .where("price", "<", maxPrice);
+      res.status(200).json(cheapMeals);
+
+    } else if ('availableReservations' in hasQuery) {
+      const reservation = hasQuery.availableReservations;
+      const freeSpots = await knex("meal")
+        .select("title", "max_reservations", knex.raw("SUM(number_of_guests)"))
+        .leftJoin("reservation", "meal.id", "=", "reservation.meal_id")
+        .groupBy("title", "max_reservations")
+        .havingRaw("max_reservations - IFNULL(SUM(number_of_guests), 0) > 0");
+
+      const noSpots = await knex("meal")
+        .select("title", "max_reservations", knex.raw("SUM(number_of_guests)"))
+        .leftJoin("reservation", "meal.id", "=", "reservation.meal_id")
+        .groupBy("title", "max_reservations")
+        .havingRaw("max_reservations - IFNULL(SUM(number_of_guests), 0) <= 0");
+
+      reservation === "true"
+        ? res.status(200).json(freeSpots)
+        : res.status(200).json(noSpots);
+
+    } else if ('title' in hasQuery) {
+      const match = hasQuery.title;
+      const matchMeal = await knex("meal")
+        .where("title", "like", `%${match}%`);
+        res.status(200).json(matchMeal)
+    }
+
+    else if ('dateAfter' in hasQuery) {
+      const dateAfter = hasQuery.dateAfter;
+      console.log(dateAfter)
+      const dateAfterMeals = await knex("meal")
+        .where("when", ">", dateAfter);
+        res.status(200).json(dateAfterMeals);
+    }
+
+    else if ('dateBefore' in hasQuery) {
+      const dateBefore = hasQuery.dateBefore;
+      console.log(dateBefore)
+      const dateBeforeMeals = await knex("meal")
+        .where("when", "<", dateBefore);
+        res.status(200).json(dateBeforeMeals);
+    }
+
+    else if ('limit' in hasQuery) {
+      const limitedNumber = hasQuery.limit;
+      console.log(limitedNumber)
+      const limitedMeals = await knex("meal")
+        .limit(limitedNumber);
+        res.status(200).json(limitedMeals);
+    }
+
+    else if ('sortKey' in hasQuery && !('sortDir' in hasQuery)) {
+      const sortKey = hasQuery.sortKey;
+      const sortedMeals = await knex("meal")
+        .orderBy(`${sortKey}`, 'asc');
+        
+      sortKey === 'when' || sortKey === 'price' || sortKey === 'max_reservations'
+        ? res.status(200).json(sortedMeals)
+        : res.status(404).json(`not possible to sort by ${sortKey}`)
+    }
+
+    else if ('sortKey' in hasQuery && 'sortDir' in hasQuery) {
+      const column = hasQuery.sortKey;
+      const direction = hasQuery.sortDir;
+      const sortedMeals = await knex("meal").
+         orderBy(`${column}`, `${direction}`)
+
+      res.status(200).json(sortedMeals)
+    }
+
+    else {
+      const allMeals = await knex("meal").select("title", "price");
+      res.status(200).json(allMeals);
+    }
```

Comment:
Overall, you have put in good efforts. And yes, you have handled every filter separately. But the scenario is, what if the user want to filter the meals with multiple filters. So there, I would suggest refactoring the code as:
- add `let query = knex('meal').select('*')`  at the top, before the if conditions for filtering.
- remove `else if` instead use all `if` conditions, because you will need to check for all filters.
- execute the query at the end of if conditions